### DISCUSSION TITLE
Windows Platform Support

### DISCUSSION THREAD

**Discussion by Orgbrat** [CommunityToolkit/Maui#181] [Q&A] [answered]
First let me say that everyone in the maintainers group of developers are the absolute very best of folks. I know I have asked this before and I was told  Windows support was coming but we were waiting on the Windows team. I also know that this has been a massive undertaking so far and will be for a while longer. But this toolkit is useless to me without Windows support. MacOs would be nice surprise but Windows is a necessity.  I am trying to replace a Xamarin.Forms (iOS, Droid, UWP) app with a Maui application. I understand all the issues you folks are working with, because as of right now for Maui things that work in Android may not work in iOS or may not work in Windows. I would really like to be able to use the Community Toolkit for Maui just as I do the toolkit in Xamarin.Forms but right now without Windows support that is not possible.

I am sorry I just needed to whine a little to someone that cares...  :)

Steve

**Comment by TheCodeTraveler** [comment, marked as answer]
Thank you for the kind words!

There's a couple questions to answer  in this discussion:

### Will .NET MAUI Community Toolkit Support Windows?

Yes. ~We will add windows support before the .NET MAUI Toolkit GAs alongside .NET MAUI in Q2 2022.~

**Update:** Windows support now available: https://github.com/CommunityToolkit/Maui/pull/188

### Does .NET MAUI Community Toolkit Support macOS?

Yes. macOS support has been available since the first pre-release of `CommunityToolkit.Maui`. 

### What Workarounds Are Available?

You can use the .NET MAUI-Compatible version of the Xamarin.CommunityToolkit, `Xamarin.CommunityToolkit.MauiCompat`: https://www.nuget.org/packages/Xamarin.CommunityToolkit.MauiCompat/

You can find more information about `Xamarin.CommunityToolkit.MauiCompat` here: https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit/?WT.mc_id=mobile-0000-bramin

### Should You Port Production Apps to .NET MAUI Today?

Today, no. In the future, yes. 

The .NET MAUI team does not recommend porting your production applications to .NET before it GAs in Q2 2022:
> Unfortunately, .NET MAUI will not be ready for production with .NET 6 GA in November. We want to provide the best experience, performance, and quality on day 1 to our users and to do that, we need to slip the schedule. We are now targeting early Q2 of 2022 for .NET MAUI GA.

https://devblogs.microsoft.com/dotnet/update-on-dotnet-maui/?WT.mc_id=mobile-0000-bramin

**Reply by Orgbrat** [reply]
@brminnick  You are the best. You know I know all this and I am not going to be ready for production until MAUI GA's in Q2 2022 .  I would like to get started so that I would not be behind  and could have  a MAUI application ready soon after Maui GA's.   And besides I have helped MAUI find some issues.  Each preview gets better and better but you folks are correct. It isn't there yet...   :)

**Reply by TheCodeTraveler** [reply]
Good news! We just added Windows Support: https://github.com/CommunityToolkit/Maui/pull/188 🎉 

The next release will support Windows 👍

**Comment by Orgbrat** [comment]
@brminnick  Thank you very very much.. You are much appreciated.  🥇

**Comment by CliffAgius** [comment]
Closed as answered
